File: kotlin-multiplatform/src/commonMain/kotlin/com/ricoh360/thetableclient/service/data/values/NetworkType.kt
package com.ricoh360.thetableclient.service.data.values

/**
 * Network type
 */
enum class NetworkType(internal val value: String?) {
    /**
     * Undefined value
     */
    UNKNOWN(null),

    /**
     * Network is off. This value can be gotten only by plugin
     */
    OFF("OFF"),

    /**
     * Direct mode
     */
    DIRECT("AP"),

    /**
     * Client mode via WLAN
     */
    CLIENT("CL"),

    /**
     * Client mode via Ethernet cable
     */
    ETHERNET("ETHERNET"),

    /**
     * LTE plan-D
     */
    LTE_PLAN_D("LTE plan-D"),

    /**
     * LTE plan-DU
     */
    LTE_PLAN_DU("LTE plan-DU"),

    /**
     * LTE plan01s
     */
    LTE_PLAN_01S("LTE plan01s"),

    /**
     * LTE planX3
     */
    LTE_PLAN_X3("LTE planX3"),

    /**
     * LTE planP1
     */
    LTE_PLAN_P1("LTE planP1"),

    /**
     * LTE plan-K2
     */
    LTE_PLAN_K2("LTE plan-K2"),

    /**
     * LTE plan-K
     */
    LTE_PLAN_K("LTE plan-K"),

    /**
     * SSID scan mode
     *
     * Switch to CL mode and search for SSID
     */
    SCAN("SCAN"),
    ;

    companion object {
        /**
         * Options property key name
         */
        val keyName: String
            get() = "networkType"

        /**
         * Search by json value.
         *
         * @param value json value.
         * @return NetworkType
         */
        fun getFromValue(value: String): NetworkType {
            return entries.firstOrNull { it.value == value } ?: run {
                println("Unknown value ${UNKNOWN::class.simpleName}: $value")
                UNKNOWN
            }
        }
    }
}

File: react-native/src/service/values/network-type.ts
/**
 * Network type of the camera.
 * Can be acquired by camera.getOptions and set by camera.setOptions.
 */
export const NetworkTypeEnum = {
  /** Undefined value */
  UNKNOWN: 'UNKNOWN',
  /** Direct mode */
  DIRECT: 'DIRECT',
  /** Client mode via WLAN */
  CLIENT: 'CLIENT',
  /** Client mode via Ethernet cable */
  ETHERNET: 'ETHERNET',
  /** Network is off. This value can be gotten only by plugin. */
  OFF: 'OFF',
  /** LTE plan-D */
  LTE_PLAN_D: 'LTE_PLAN_D',
  /** LTE plan-DU */
  LTE_PLAN_DU: 'LTE_PLAN_DU',
  /** LTE plan01s */
  LTE_PLAN_01S: 'LTE_PLAN_01S',
  /** LTE planX3 */
  LTE_PLAN_X3: 'LTE_PLAN_X3',
  /** LTE planP1 */
  LTE_PLAN_P1: 'LTE_PLAN_P1',
  /** LTE plan-K2 */
  LTE_PLAN_K2: 'LTE_PLAN_K2',
  /** LTE plan-K */
  LTE_PLAN_K: 'LTE_PLAN_K',
  /**
   * SSID scan mode
   * 
   * Switch to CL mode and search for SSID
   */
  SCAN: 'SCAN',
} as const;

/** Type definition of NetworkTypeEnum */
export type NetworkTypeEnum =
  (typeof NetworkTypeEnum)[keyof typeof NetworkTypeEnum];
